core/services/pipeline/task_params.go

```diff
@@ -717,7 +717,23 @@ func (s *AddressSliceParam) UnmarshalPipelineParam(val interface{}) error {
 
 type JSONPathParam []string
 
+// NewJSONPathParam returns a new JSONPathParam using the given separator, or the default if empty.
+func NewJSONPathParam(sep string) JSONPathParam {
+	if len(sep) == 0 {
+		return nil
```

Comment:
Will line 733 panic if `p` is `nil`?
I know this is very abnormal case, but since the constructor can return `nil`...

Reply:
Possibly? Or maybe not since there is another layer of indirection through the slice? Either way, I'll add a test to cover this.

Reply:
I am referring to dereferencing: `*p`. FYI.